artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/AddressSettings.java

```diff
@@ -759,6 +761,15 @@ public AddressSettings setDefaultRingSize(final long defaultRingSize) {
       return this;
    }
 
+   public long getRetroactiveMessageCount() {
+      return retroactiveMessageCount != null ? retroactiveMessageCount : ActiveMQDefaultConfiguration.DEFAULT_RETROACTIVE_MESSAGE_COUNT;
+   }
+
+   public AddressSettings setRetroactiveMessageCount(final long defaultRetroactiveMessageCount) {
+      this.retroactiveMessageCount = defaultRetroactiveMessageCount;
+      return this;
+   }
+
```

Comment:
I have tried to add a comment on a line you didn't touch on github and I couldn't  figure out how to do it...


It seems you missed implementing the merge method here,

you also missed implementing the persistence of the new attribute here.


So, I think you should add implementation on this new attribute, and you should probably also add tests on loading the new setting.